lib/panda_motd/components/ssl_certificates.rb

```diff
@@ -11,56 +11,73 @@ def process
-          date_portion = cert[1].strftime('%e %b %Y %H:%M:%S%p')
-          "  #{name_portion} #{status} #{date_portion}"
+          parse_cert(cert)
         end.join("\n")}
     HEREDOC
   end
 
   private
 
+  def parse_cert(cert)
+    name_portion = cert[0].ljust(longest_cert_name_length + 6, ' ')
+    status_sym = cert_status(cert[1])
+    status = cert_status_strings[status_sym].to_s
+    colorized_status = status.colorize(cert_status_colors[status_sym])
+    date_portion = cert[1].strftime('%e %b %Y %H:%M:%S%p')
+    "  #{name_portion} #{colorized_status} #{date_portion}"
+  end
+
+  def longest_cert_name_length
+    @results.map { |r| r[0].length }.max
+  end
+
+  def sorted_results
+    if @config['sort_method'] == 'alphabetical'
+      @results.sort_by { |c| c[0] }
+    elsif @config['sort_method'] == 'expiration'
+      @results.sort_by { |c| c[1] }
+    else # default to alphabetical
+      @results.sort_by { |c| c[0] }
+    end
+  end
+
   def cert_dates(certs)
-    return certs.map do |name, path|
+    certs.map do |name, path|
       if File.exist?(path)
-        cmd_result = `openssl x509 -in #{path} -dates`
-        # match indices: 1 - month, 2 - day, 3 - time, 4 - year, 5 - zone
-        parsed = cmd_result.match(/notAfter=([A-Za-z]+) (\d+) ([\d:]+) (\d{4}) ([A-Za-z]+)\n/)
-        if parsed.nil?
-          @errors << ComponentError.new(self, 'Unable to find certificate expiration date')
-          nil
-        else
-          begin
-            expiry_date = Time.parse("#{parsed[1]} #{parsed[2]} #{parsed[4]} #{parsed[3]} #{parsed[5]} ")
-            [name, expiry_date]
-          rescue ArgumentError
-            @errors << ComponentError.new(self, 'Found expiration date, but unable to parse as date')
-            [name, Time.now]
-          end
-        end
+        parse_result(name, path)
       else
         "Certificate #{name} not found at path: #{path}"
       end
     end.compact # remove nil entries, will have nil if error ocurred
   end
 
+  def parse_result(name, path)
+    cmd_result = `openssl x509 -in #{path} -dates`
+    # match indices: 1 - month, 2 - day, 3 - time, 4 - year, 5 - zone
+    exp = /notAfter=([A-Za-z]+) (\d+) ([\d:]+) (\d{4}) ([A-Za-z]+)\n/
+    parsed = cmd_result.match(exp)
+
+    if parsed.nil?
+      @errors << ComponentError.new(self, 'Unable to find certificate '\
+                                          'expiration date')
+      nil
+    else
+      expiry_date = Time.parse([1, 2, 4, 3, 5].map { |n| parsed[n] }.join(' '))
+      [name, expiry_date]
+    end
+  rescue ArgumentError
+    @errors << ComponentError.new(self, 'Found expiration date, but unable '\
+                                        'to parse as date')
+    [name, Time.now]
+  end
+
   def cert_status(expiry_date)
-    if (Time.now...Time.now + 30).cover? expiry_date # ... range excludes end
+    if (Time.now...Time.now + 30).cover? expiry_date # ... range excludes last
```

Comment:
Did this because tpope/vim-endwise#113. Took a couple hours to figure out why it wasn't working...